graph/src/schema/input_schema.rs

```diff
@@ -79,11 +81,13 @@ impl InputSchema {
                         .iter()
                         .map(|field| pool.lookup(&field.name).unwrap())
                         .collect();
-                    (EntityType::from(obj_type), fields)
-                }),
+                    EntityType::new(&pool, &obj_type.name).map(|t| (t, fields))
+                })
+                .collect::<Result<Vec<_>, _>>()
+                .unwrap(),
```

Comment:
if this unwrap is just because EntityType::new can't fail then I think it's probably better to just unwrap the new, similar to how the lookup is also unwrapped, IMO it would make it slightly clearer even though it still makes me a bit nervous to see all the unwraps 😅